lib/ansible/module_utils/common/dict_transformations.py

```diff
@@ -123,6 +123,8 @@ def dict_merge(a, b):
 
 
 def recursive_diff(dict1, dict2):
+    if not isinstance(dict1, dict) or not isinstance(dict2, dict):
```

Comment:
It's a bit more flexible use `MutableMapping` for instance tests. We have this in `ansible.module_utils.common._collections_compat` since it moved between PY2/3.
```suggestion
    if not isinstance(dict1, MutableMapping) or not isinstance(dict2, MutableMapping):
```

Reply:
👍🏼